pkg/cmd/diff/diff.go

```diff
@@ -0,0 +1,127 @@
+// Copyright 2024 The KitOps Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package diff
+
+import (
+	"context"
+	"fmt"
+	"sort"
+	"strings"
+
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+	"oras.land/oras-go/v2/registry"
+
+	"kitops/pkg/lib/constants"
+	"kitops/pkg/lib/repo/local"
+	"kitops/pkg/lib/repo/remote"
+	"kitops/pkg/lib/repo/util"
+)
+
+// Helper struct diffInfo holds the manifest and its descriptor for a ModelKit.
+type diffInfo struct {
+	Manifest   *ocispec.Manifest
+	Descriptor ocispec.Descriptor
+}
+
+// Helper struct diffResult contains the comparison results between two ModelKits.
+type diffResult struct {
+	SharedLayers  []ocispec.Descriptor
+	UniqueLayersA []ocispec.Descriptor
+	UniqueLayersB []ocispec.Descriptor
+}
+
+// compareManifests compares two OCI manifests and returns the shared and unique layers.
+func compareManifests(manifestA *ocispec.Manifest, manifestB *ocispec.Manifest) *diffResult {
```

Comment:
There are two pieces that we're not diffing that could lead to confusing results:

* Any annotations on the manifest (which will happen if e.g. the modelkit is packed by a different version of kit)
* The config for the ModelKit (which could point at e.g. different parameters on the same model data)